logdna/resource_view_test.go

```diff
@@ -246,6 +275,128 @@ func TestViewBulkEmails(t *testing.T) {
 	})
 }
 
+func TestViewBulkEmailsUpdate(t *testing.T) {
+	name := "test"
+	query := "test"
+	app1 := "app1"
+	app2 := "app2"
+	levels1 := "fatal"
+	levels2 := "critical"
+	host1 := "host1"
+	host2 := "host2"
+	category := "DEMO"
+	category2 := "DEMO2"
+	tags1 := "host1"
+	tags2 := "host2"
+
+	name2 := "test2"
+	query2 := "query2"
+	app3 := "app3"
+	app4 := "app4"
+	levels3 := "error"
+	levels4 := "warning"
+	host3 := "host3"
+	host4 := "host4"
+	category3 := "DEMO3"
+	category4 := "DEMO4"
+	tags3 := "tags3"
+	tags4 := "tags4"
+
+	resource.Test(t, resource.TestCase{
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testViewConfigBulkEmails(name, query, app1, app2, levels1, levels2, host1, host2, category, category2, tags1, tags2),
```

Comment:
This pattern of adding infinite function parameters to a templating seems pretty unmaintainable.  The test config is very static, even for the fields that are using the values passed in - there are always 2 apps, 2 hosts, etc. So the added indirection of a function wrapper around `fmt.Sprintf` doesn't really do much for us, aside from providing a place to store the template string. I don't think we're gaining much here by having this as a function vs just defining the template string as a top-level variable and calling `fmt.Sprintf` directly.

I don't think it's necessary to change in this PR, just a pattern I think we should find an alternative for in future work.

Reply:
how should we try to change this in the future? passing along the values in the sprintf also makes sense in the context that I can assert that the values match the variables in the TestCheckResourcesAttr but I also do agree that the # of variables is a lot for a couple of the tests

Reply:
i could not sprintf and pass anything to the config files but still have the variables as reference and assert on them in the checkresourcesattr

Reply:
yeah you could have the template string as a top-level variable and use `fmt.Sprintf` where you need it. just removing a layer:


```golang
var testConfigBulkEmails = `provider "logdna" {
	}
	resource "logdna_view" "new" {
		name = "%s"
		query = "%s"
	}`

func TestViewBulkEmailsUpdate(t *testing.T) {
    foo := "foo"
    bar := "bar"
    
    config := fmt.Sprintf(testConfigBulkEmails, foo, bar)
    ...
```
there might be other ways to manage all of this too, I'm just thinking of the simplest form.